### PR TITLE
add com.blockstream.Green

### PR DESCRIPTION
**Pull request by grubles**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
~~`--filesystem=home:rw` is used to give the flatpak access to the user's home directory to store logs and other app-related files.~~
`--device=all` is required to connect to serial devices that can be used with the app.
`--share=network` is required so the app can connect to remote servers.
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.blockstream.Green.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    build-commands:
+      - bsdtar -xf BlockstreamGreen_Linux_x86_64.tar.gz
+      - install -D green ${FLATPAK_DEST}/bin/green
+      - desktop-file-install --dir ${FLATPAK_DEST}/share/applications com.blockstream.Green.desktop
+      - install -Dm644 green.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.blockstream.Green.png
+      - install -Dm644 com.blockstream.Green.metainfo.xml ${FLATPAK_DEST}/share/metainfo/com.blockstream.Green.metainfo.xml
+
+    sources:
+      - type: file
+        url: https://github.com/Blockstream/green_qt/releases/download/release_1.2.0/BlockstreamGreen_Linux_x86_64.tar.gz
```
Comment (hfiguiere) [inline com.blockstream.Green.yml:18]: so how is this gonna work on aarch64?

File: com.blockstream.Green.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    buildsystem: simple
+    build-commands:
+      - bsdtar -xf BlockstreamGreen_Linux_x86_64.tar.gz
+      - install -D green ${FLATPAK_DEST}/bin/green
+      - desktop-file-install --dir ${FLATPAK_DEST}/share/applications com.blockstream.Green.desktop
+      - install -Dm644 green.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.blockstream.Green.png
+      - install -Dm644 com.blockstream.Green.metainfo.xml ${FLATPAK_DEST}/share/metainfo/com.blockstream.Green.metainfo.xml
+
+    sources:
+      - type: file
```
Comment (hfiguiere) [inline com.blockstream.Green.yml:17]: you should use the archive type instead and let flatpak extract the archive.

File: com.blockstream.Green.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		  </description>
+	  </release>
+  </releases>
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://blockstream.com/img/img/laptop_green_ui_lander.png</image>
+    </screenshot>
+  </screenshots>
+</component>
```
Comment (hfiguiere) [inline com.blockstream.Green.metainfo.xml:107]: missing required content_rating.

File: com.blockstream.Green.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+sdk: org.kde.Sdk
+command: green
+modules: 
+  - name: green
+    buildsystem: simple
+    build-commands:
+      - bsdtar -xf BlockstreamGreen_Linux_x86_64.tar.gz
+      - install -D green ${FLATPAK_DEST}/bin/green
+      - desktop-file-install --dir ${FLATPAK_DEST}/share/applications com.blockstream.Green.desktop
+      - install -Dm644 green.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.blockstream.Green.png
```
Comment (hfiguiere) [inline com.blockstream.Green.yml:13]: the PNG isn't 256x256

File: com.blockstream.Green.yml (hunk 1 above)
Comment (grubles) [reply com.blockstream.Green.yml:18]: It won't. We only provide binaries for x86_64 currently.

Comment (hfiguiere) [reply com.blockstream.Green.yml:18]: you can build from source. As you currently still try to make an aarch64 flatpak

**Comment by hfiguiere** [comment]
have the .desktop and appstream file been submitted upstream?

File: com.blockstream.Green.yml (hunk 1 above)
Comment (grubles) [reply com.blockstream.Green.yml:18]: Aarch64 support isn't production ready yet.

**Comment by grubles** [comment]
The appstream file has not been upstreamed yet. The `.desktop` is already [upstream](https://github.com/Blockstream/green_qt/blob/master/green.desktop).

File: com.blockstream.Green.metainfo.xml (hunk 3 above)
Comment (grubles) [reply com.blockstream.Green.metainfo.xml:113]: Added

File: com.blockstream.Green.yml (hunk 4 above)
Comment (grubles) [reply com.blockstream.Green.yml]: Oops. Fixed.

File: com.blockstream.Green.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        sha256: c5d093a7a630ed480aafeec91289f00150eb90423c2928f8a47b8db2d9440079
+
+      - type : file
+        path: com.blockstream.Green.desktop
+
+      - type: file
+        path: com.blockstream.Green.metainfo.xml
+
+
+finish-args:
```
[the remark this answers is not in the document]
Comment (grubles) [reply com.blockstream.Green.yml]: Moved them above modules.

File: com.blockstream.Green.yml (hunk 2 above)
Comment (grubles) [reply com.blockstream.Green.yml]: Ok removed the `bsdtar` command.

**Comment by grubles** [comment]
Removed the `filesystem=home:rw` perms as they are actually unnecessary!

File: com.blockstream.Green.desktop
Comment (hfiguiere) [inline com.blockstream.Green.desktop:1]: if it is upstream this should be downloaded from upstream.

**Comment by hfiguiere** [comment]
bot, build com.blockstream.Green

File: com.blockstream.Green.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: green
+
+finish-args:
+  - --share=network
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=ipc
+  - --talk-name=org.freedesktop.portal.Background
```
Comment (hfiguiere) [inline com.blockstream.Green.yml:12]: I do believe this is unecessary

File: com.blockstream.Green.yml (hunk 1 above)
Comment (hfiguiere) [reply com.blockstream.Green.yml:26]: you should still build from source

File: com.blockstream.Green.yml (hunk 6 above)
Comment (grubles) [reply com.blockstream.Green.yml:12]: Which part?

Comment (hfiguiere) [reply com.blockstream.Green.yml:12]: talk-name that's the line I commented on

File: com.blockstream.Green.desktop
Comment (grubles) [reply com.blockstream.Green.desktop:1]: Fixed

File: com.blockstream.Green.yml (hunk 6 above)
Comment (grubles) [reply com.blockstream.Green.yml]: You're right. Tested without it. I've removed it from the manifest.

**Comment by hfiguiere** [comment]
bot, build com.blockstream.Green

**Review by hfiguiere — CHANGES_REQUESTED** [review]
appstream fails to validate.

you still make an aarch64 flatpak out of x86_64 binaries.

**Comment by grubles** [comment]
```
builddir/files/share/appdata/com.blockstream.Green.appdata.xml: FAILED:
• style-invalid         : Too many <p> tags for a good description [24/10]
```

Too many `<p>` tags in `com.blockstream.Green.metainfo.xml` ?

**Comment by grubles** [comment]
> appstream fails to validate.
> 
> you still make an aarch64 flatpak out of x86_64 binaries.

Ok it looks like we need some combination of `only-arches` and/or `skip-arches` then?

**Comment by grubles** [comment]
bot, build com.blockstream.Green

**Comment by grubles** [comment]
bot, build com.blockstream.Green

**Comment by grubles** [comment]
Can you add @wtogami as a collaborator?

**Comment by grubles** [comment]
@barthalion See above. Ty.

**Comment by grubles** [comment]
@hfiguiere Are you able to add @wtogami to flathub/com.blockstream.Green as a collaborator?

**Comment by hfiguiere** [comment]
no

file an issue https://github.com/flathub/flathub/issues

**Pull request closed without merging**